pkgs/development/libraries/vectorscan/default.nix

```
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ragel
+, util-linux
+, python3
+, boost
+, enableShared ? !stdenv.hostPlatform.isStatic
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vectorscan";
+  version = "5.4.9";
+
+  src = fetchFromGitHub {
+    owner = "VectorCamp";
+    repo = "vectorscan";
+    rev = "vectorscan/${version}";
+    hash = "sha256-V5Qgr8aH1H+ZoJ0IZ52HIDRZq+yIwHjLf3gU/ZhjjlY=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ragel
+    util-linux
+    python3
+  ];
+
+  buildInputs = [
+    boost
+  ];
+
+  cmakeFlags = lib.optional (enableShared) [
+    "-DBUILD_STATIC_AND_SHARED=ON"
+  ] ++ lib.optional (stdenv.hostPlatform.system == "x86_64-linux") [
+    "-DBUILD_AVX2=ON"
+    "-DBUILD_AVX512=ON"
+    "-DFAT_RUNTIME=ON"
+  ];
+
+  meta = with lib; {
+    description = "A portable fork of the high-performance regular expression matching library";
+    longDescription = ''
+      A fork of Intel's Hyperscan, modified to run on more platforms. Currently
+      ARM NEON/ASIMD is 100% functional, and Power VSX are in development.
+      ARM SVE2 will be implemented when hardware becomes accessible to the
+      developers. More platforms will follow in the future, on demand/request.
+
+      Vectorscan will follow Intel’s API and internal algorithms where possible,
+      but will not hesitate to make code changes where it is thought of giving
+      better performance or better portability. In addition, the code will be
+      gradually simplified and made more uniform and all architecture specific
+      code will be abstracted away.
+    '';
+    homepage = "https://www.vectorcamp.gr/vectorscan/";
+    changelog = "https://github.com/VectorCamp/vectorscan/blob/${src.rev}/CHANGELOG.md";
+    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
```

```suggestion
    platforms = platforms.unix;
```
more or less for our purposes

---

I tested with i686-linux and it does not work. I assume, that many platforms in platforms.unix are not suppored by vectorscan. Should I still use platforms.unix?

---

The darwin builds in ofborg fail. => Only x86_64 and aarch64 left.

---

This is more or less about what hydra should build. Most programs that have unix probably don't support everything in there and that is alright.

If darwin does not work, we should set this to platforms.linux. If also aarch64 does not work, we should set it to ``[ "x86_64-linux" ]``

---

Darwin should work, given that vectorscan is available

---

Darwin should work, as I have binary wheels targeting that platform. I am unsure what the issue is

---

From the [logs](https://logs.ofborg.org/?attempt_id=6f3aebce-da7b-4f31-b526-899927644f1a&key=nixos%2Fnixpkgs.232314) I think fat runtime defaults to enabled, this causes the build errors IMHO
I explicitly disable it on darwin, when build statically: https://github.com/vlaci/pyperscan/blob/e8073597ef9532878d9209acf68c97f5a7433985/hyperscan-sys/build.rs#L84

---

Darwin now works and we have platforms.unix. => resolving this thread